mimicpy/parsers/top.py

```diff
@@ -12,13 +12,16 @@ def fromDict(cls, df):
         keys = list(df.keys())
         df2 = df.copy()
         repeating = {}
-        for i in range(len(keys)):
+        i = 0
+        while i < len(keys):
             key_i = keys[i]
             for j in range(i+1, len(keys)):
                 key_j = keys[j]
                 if df[key_i].equals(df[key_j]):
                     repeating[key_j] = key_i
                     del df2[key_j]
+            i += 1
+            keys = list(df2.keys())
```

Comment:
Ah, see previous comment. Does python check len(some_list) every iteration of the while loop or just once at the first time?
I think I would prefer readability over performance here. Or is it so much faster?

Reply:
This is not a question of performance. It simply doesn't work with a for loop. However, there might be a prettier way to do this. We can discuss that.